lib/logidze/meta.rb

```diff
@@ -50,30 +79,46 @@ def call_block_in_meta_context
       ensure
         meta_stack.pop
       end
+    end
 
-      def current_meta
-        meta_stack.reduce(:merge) || {}
-      end
+    class MetaWithoutTransaction < MetaWrapper # :nodoc:
+      def perform
+        raise ArgumentError, "Block must be given" unless block
 
-      def meta_stack
-        Thread.current[:meta] ||= []
-        Thread.current[:meta]
+        call_block_in_meta_context
       end
 
+      private
+
       def pg_set_meta_param(value)
-        encoded_meta = connection.quote(ActiveSupport::JSON.encode(value))
-        connection.execute("SET LOCAL logidze.meta = #{encoded_meta};")
+        connection.execute("SET logidze.meta = #{encode_meta(value)};")
       end
 
       def pg_reset_meta_param(prev_meta)
         if prev_meta.empty?
-          connection.execute("SET LOCAL logidze.meta TO DEFAULT;")
+          connection.execute("SET logidze.meta TO DEFAULT;")
```

Comment:
Let's extract this into a  `pg_clear_meta_param` method and move `pg_reset_meta_param` to the base class.. Thus, we'll only have to override two methods here.